dgp/lib/transform/derivatives.py

```diff
@@ -17,44 +17,17 @@ def centraldifference(data_in, n=1, order=2, dt=0.1):
         else:
             raise ValueError('Invalid value for parameter n {1 or 2}')
     else:
-        raise NotImplementedError()
+        raise NotImplementedError
 
     return np.pad(dy, (1, 1), 'edge')
-    return dy
-
-
-def gradient(data_in, dt=0.1):
-    return np.gradient(data_in, dt)
-
 
-class CentralDifference(CtrlNode):
-    nodeName = "centraldifference"
-    uiTemplate = [
-        ('order', 'combo', {'values': [2, 4], 'index': 0}),
-        ('n', 'combo', {'values': [1, 2], 'index': 0}),
-        ('dt', 'spin', {'value': 0.1, 'step': 0.1, 'bounds': [0.0001, None]})
-    ]
-
-    def __init__(self, name):
-        terminals = {
-            'data_in': dict(io='in'),
-            'data_out': dict(io='out'),
-        }
-
-        CtrlNode.__init__(self, name, terminals=terminals)
-
-    def process(self, data_in, display=True):
-        if self.ctrls['order'] == 2:
-            # first derivative
-            if self.ctrls['n'] == 1:
-                dy = (data_in[2:] - data_in[0:-2]) / (2 * self.ctrls['dt'])
-            # second derivative
-            elif self.ctrls['n'] == 2:
-                dy = ((data_in[0:-2] - 2 * data_in[1:-1] + data_in[2:]) /
-                      np.power(self.ctrls['dt'], 2))
-            else:
-                raise ValueError('Invalid value for parameter n {1 or 2}')
-        else:
-            raise NotImplementedError()
 
-        return {'data_out': np.pad(dy, (1, 1), 'edge')}
\ No newline at end of file
+# TODO: Add option to specify order
+def taylor_fir(data_in, n=1, dt=0.1):
+    """ 10th order Taylor series FIR differentiator """
+    coeff = np.array([1 / 1260, -5 / 504, 5 / 84, -5 / 21, 5 / 6, 0, -5 / 6, 5 / 21, -5 / 84, 5 / 504, -1 / 1260])
+    x = data_in
+    for _ in range(1, n + 1):
```

Comment:
Should be able to remove 'y' variable and replace with
```
x = convolve(x, coeff, mode='same')
```
Fixes linter warning about y 'might be referenced before assignment'

Reply:
Done. In fact, made it even a bit cleaner. I don't frankly know why I did it that way.